internal/controller/mariadb_controller_replica_recovery_test.go

```diff
@@ -114,3 +123,259 @@ var _ = Describe("isRecoverableError", func() {
 		),
 	)
 })
+
+var _ = Describe("MariaDB Replica Recovery", Ordered, func() {
+	var (
+		key = types.NamespacedName{
+			Name:      "mariadb-repl",
+			Namespace: testNamespace,
+		}
+		mdb *mariadbv1alpha1.MariaDB
+	)
+
+	BeforeEach(func() {
+		mdb = buildTestMariaDBRecovery(key)
+		applyMariadbTestConfig(mdb)
+
+		By("Creating MariaDB with replication")
+		Expect(k8sClient.Create(testCtx, mdb)).To(Succeed())
+		DeferCleanup(func() {
+			deleteMariadb(key, false)
+		})
+
+		By("Expecting MariaDB to be ready eventually")
+		Eventually(func() bool {
+			if err := k8sClient.Get(testCtx, key, mdb); err != nil {
+				return false
+			}
+			return mdb.IsReady()
+
+		}, testHighTimeout, testInterval).Should(BeTrue())
+	})
+
+	DescribeTable(
+		"should recover",
+		func(
+			backupKey types.NamespacedName,
+			builderFn physicalBackupBuilder,
+			cleanupFn func(backupKey types.NamespacedName) func(),
+		) {
+			podIndexToDelete := 2
+			backup := builderFn(backupKey)
+			backup.Spec.Schedule = &mariadbv1alpha1.PhysicalBackupSchedule{
+				Suspend: true,
+			}
+			Expect(k8sClient.Create(testCtx, backup)).To(Succeed())
+			DeferCleanup(func() {
+				Expect(client.IgnoreNotFound(k8sClient.Delete(testCtx, backup))).To(Succeed())
+			})
+
+			DeferCleanup(func() {
+				deletePhysicalBackup(backupKey)
+				cleanupFn(backupKey)()
+			})
+
+			By("Bootstrapping Recovery")
+			Eventually(func() bool {
```

Comment:
What is the motivation of this `Eventually` block? Did you get any error updating the `MariaDB` resource? At this point we are just updating the resource, not triggering the recovery.